pkg/envoy/lds/ingress.go

```diff
@@ -9,66 +9,59 @@ import (
 	"google.golang.org/protobuf/types/known/anypb"
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
-	configv1alpha2 "github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
-
 	"github.com/openservicemesh/osm/pkg/constants"
 	"github.com/openservicemesh/osm/pkg/envoy"
 	"github.com/openservicemesh/osm/pkg/envoy/rds/route"
 	"github.com/openservicemesh/osm/pkg/errcode"
-	"github.com/openservicemesh/osm/pkg/service"
 	"github.com/openservicemesh/osm/pkg/trafficpolicy"
 )
 
-func (lb *listenerBuilder) getIngressFilterChains(svc service.MeshService) []*xds_listener.FilterChain {
-	ingressPolicy, err := lb.meshCatalog.GetIngressTrafficPolicy(svc)
-	if err != nil {
-		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrIngressFilterChain)).
-			Msgf("Error getting ingress filter chain for proxy with identity %s and service %s", lb.serviceIdentity, svc)
-		return nil
-	}
-
-	if ingressPolicy == nil {
-		log.Trace().Msgf("No ingress policy confiugred for proxy with identity %s and service %s", lb.serviceIdentity, svc)
+func (lb *listenerBuilder) buildIngressFilterChains() []*xds_listener.FilterChain {
+	if lb.ingressTrafficPolicies == nil {
 		return nil
 	}
 
 	var filterChains []*xds_listener.FilterChain
-	for _, trafficMatch := range ingressPolicy.TrafficMatches {
-		if filterChain, err := lb.getIngressFilterChainFromTrafficMatch(trafficMatch, lb.cfg.GetMeshConfig().Spec.Sidecar); err != nil {
-			log.Error().Err(err).Msgf("Error building ingress filter chain for proxy with identity %s service %s", lb.serviceIdentity, svc)
-		} else {
+
+	for _, policy := range lb.ingressTrafficPolicies {
+		for _, trafficMatch := range policy.TrafficMatches {
+			filterChain, err := lb.buildIngressFilterChainFromTrafficMatch(trafficMatch)
+			if err != nil {
+				log.Error().Err(err).Msgf("Error building ingress filter chain for traffic match %s for proxy with identity %s", trafficMatch.Name, lb.proxyIdentity)
```

Comment:
This doesn't look to have changed here. But out of curiosity, Is this an expected error in some cases? what happens if this filter chain isn't there? traffic doesn't flow?

Reply:
Yes. It's an expected error case that must be logged.